core/src/main/java/org/elasticsearch/cluster/metadata/IndexMetaData.java

```diff
@@ -279,6 +286,7 @@ public static State fromString(String state) {
     private final Index index;
     private final long version;
     private final long[] primaryTerms;
+    private final int internalIndexFormat;
```

Comment:
do we really have to make this a first class citizen or can we just leave it as a setting?